lsan.supp

```diff
@@ -0,0 +1,2 @@
+leak:packet_create
+leak:add_missing_streams
```

Comment:
I don't think it's a good idea to make everyone edit their LSAN_OPTIONS env var. Instead we can just use `__lsan_default_suppressions()` to embed this into the binary.

However, for whatever reason, this doesn't seem to be working:

```diff
--- a/ta/ta.c
+++ b/ta/ta.c
@@ -71,6 +71,16 @@ static void ta_dbg_add(struct ta_header *h);
 static void ta_dbg_check_header(struct ta_header *h);
 static void ta_dbg_remove(struct ta_header *h);
 
+#if __SANITIZE_ADDRESS__
+const char *__lsan_default_suppressions(void);
+const char *__lsan_default_suppressions(void)
+{
+    return "leak:packet_create\n"
+           "leak:add_missing_streams\n";
+}
+#endif
+
 static struct ta_header *get_header(void *ptr)
 {
     struct ta_header *h = ptr ? PTR_TO_HEADER(ptr) : NULL;
```

It works fine on a small test program, so I don't think the problem is with my toolchain. I'm also not seeing anything alarming on mpv's compiler arguments either. Any ideas?

Reply:
I didn't want to force this suppression. But maybe this is actually better to avoid annoyance of changing options like you said.

> It works fine on a small test program, so I don't think the problem is with my toolchain. I'm also not seeing anything alarming on mpv's compiler arguments either. Any ideas?

Yep, need `__attribute__((visibility("default"))`, we have hidden by default.

Reply:
added